addons/medical_treatment/functions/fnc_addTrauma.sqf

```diff
@@ -0,0 +1,43 @@
+#include "..\script_component.hpp"
+/*
+ * Author: LinkIsGrim
+ * Add trauma to a body part.
+ *
+ * Arguments:
+ * 0: Patient <OBJECT>
+ * 1: Body Part <STRING>
+ * 2: Amount of trauma to add, negative to remove <NUMBER>
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [player, "head", -5] call ace_medical_treatment_fnc_addTrauma
+ *
+ * Public: No
+ */
+
+params ["_unit", "_bodyPart", "_trauma"];
+
+private _partIndex = ALL_BODY_PARTS find _bodyPart;
+private _bodyPartDamage = GET_BODYPART_DAMAGE(_unit) select _partIndex;
+private _newDam = (_bodyPartDamage select _partIndex) + _trauma;
+
+// Prevent obscenely small damage from lack of floating precision
+if (_trauma < 0 && _newDam < 0.05) then {
+    _bodyPartDamage set [_partIndex, 0];
+} else {
+    _bodyPartDamage set [_partIndex, _newDam];
+};
+
+// Update body part visuals
+switch (_bodyPart) do {
+    case "head":     { [_unit, true, false, false, false] call EFUNC(medical_engine,updateBodyPartVisuals); };
+    case "body":     { [_unit, false, true, false, false] call EFUNC(medical_engine,updateBodyPartVisuals); };
+    case "leftarm";
+    case "rightarm": { [_unit, false, false, true, false] call EFUNC(medical_engine,updateBodyPartVisuals); };
+    default { [_unit, false, false, false, true] call EFUNC(medical_engine,updateBodyPartVisuals); };
+};
+
+TRACE_2("trauma - added damage",_trauma,_newDam);
+_unit setVariable [VAR_BODYPART_DAMAGE, _bodyPartDamage, true];
```

Comment:
I'd *prefer* if this were set before handling the visuals, as the visuals reference this var, but in terms of functionality it doesn't make a difference.

Reply:
> This is just the broadcast, actual value is already modified by reference.

Yea I know, but to me to seems "safer" despite it not actually being so.

> Honestly looking at it now this function would make more sense in status I think.

You do you, but I have to fix a bug in this function, pushing right now.